### PR TITLE
add bumpversion config and maintainer's guide

### LINKED ISSUE #36: Improve release process
* add versionbump or bumpversion
* use tox to build dists and upload them
* maybe use Travis to upload tagged releases

### PR DESCRIPTION
**Pull request by leifwalsh**
Closes #36

### CODE REVIEW

**Comment by leifwalsh** [comment]
Implicit in this change is my new and weakly held opinion that we should bump all three version strings at once, rather than having distinct release cycles for each, which we can discuss.

**Comment by thejunglejane** [comment]
Okay let's talk about this. The reason I want to bump `marbles.core` and `marbles.mixins` versions independently is that they're unlikely to be changing in lock step, and changing the version when nothing has changed about the code at all feels...wrong. I do think the metapackage's version should always bump when either of `marbles.core`'s version or `marbles.mixins`'s version gets bumped.

File: Pipfile [hunk 1]
```diff
@@ -14,3 +14,4 @@ coverage = "*"
 "flake8" = "*"
 tox = "*"
 "flake8-per-file-ignores" = "*"
```
Comment (thejunglejane) [inline Pipfile]: minor: can we put `flake8-per-file-ignores` next to `flake8`?

File: docs/install.rst [hunk 2]
```diff
@@ -1,9 +1,9 @@
 Installation
 ============
 
-marbles can be installed from `PyPi`_.
+marbles can be installed from `PyPI`_.
```
Comment (thejunglejane) [inline docs/install.rst]: ty 💙

**Comment by leifwalsh** [comment]
If we want to bump the metapackage version whenever we release either subpackage, I'm not sure what its version should increase to. If we make it the max of either one, then if mixins is behind core, and we release a new version of mixins, we then wouldn't bump the metapackage. Should we then do a similar (i.e. major/minor/patch) version bump on the metapackage? That feels weird too.

Suppose we're at:

* marbles-0.10.5
* marbles.core-0.10.5
* marbles.mixins-0.9.4

If we release marbles.mixins-0.9.5, would we go to this?

* marbles-0.10.6
* marbles.core-0.10.5
* marbles.mixins-0.9.5

Or if we did marbles.mixins-0.10.0, this?

* marbles-0.11.0
* marbles.core-0.10.5
* marbles.mixins-0.10.0

I can't come up with something satisfying here. If we just increment them all in lock-step, this problem goes away.

One other argument for incrementing them all the same way is that without what feels to me like would be an incredibly complicated test suite, we can't test different combinations of the subpackages against each other. By incrementing versions all the same way, we make sure that, as long as you're using the same version of both subpackages, you're using something we've tested in CI. You'd of course be able to install different versions of each subpackage, if you wanted to, but you'd be a bit more on your own (and explicitly so, and `python -m marbles --version` would make that obvious).

File: docs/maintaining.rst [hunk 3]
```diff
@@ -170,3 +173,48 @@ make sure your clone is up to date and clean, build both ``sdist`` and
...
+    any of them gets a new release.
+
+`Jupyter <https://pypi.org/project/jupyter/>`_ takes the first
+approach, but keep in mind that Jupyter is a much larger project with
+distinct teams working on each component, so allowing subpackages to
+have independent release schedules makes more sense for that
+community.
+
+The second approach has the problem that if we release the subpackages
+independently, it's unclear what to name the metapackage when that
```
Comment (thejunglejane) [inline docs/maintaining.rst]: s/what to name/how to version

File: docs/maintaining.rst [hunk 4]
```diff
@@ -170,3 +173,48 @@ make sure your clone is up to date and clean, build both ``sdist`` and
...
+happens. Taking the max of the subpackage version strings doesn't work
+if the subpackage with a lower one gets an update by itself. There are
+a couple other possiblities here, but none of them seemed right.
+
+The third approach, of updating everything in lock-step is what we've
```
Comment (thejunglejane) [inline docs/maintaining.rst]: comma after "lock-step"

File: docs/maintaining.rst [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+
+    $ tox
+
+You can also run a subset of these with ``-e``::
+
+    $ tox -e docs
+    $ tox -e py36
+    $ tox -e flake8,coverage
+
+Releasing a new version
```
[the remark this answers is not in the document]
Comment (leifwalsh) [reply docs/maintaining.rst]: Can we leave that as an open question for now, and add it to these docs once we've decided? If you're okay with that I'll open an issue for it.

[the remark this answers is not in the document]
Comment (thejunglejane) [reply docs/maintaining.rst]: Sure

File: docs/maintaining.rst [hunk 6]
```diff
@@ -205,12 +205,12 @@ have independent release schedules makes more sense for that
...
 
 The second approach has the problem that if we release the subpackages
-independently, it's unclear what to name the metapackage when that
+independently, it's unclear how to version the metapackage when that
 happens. Taking the max of the subpackage version strings doesn't work
 if the subpackage with a lower one gets an update by itself. There are
 a couple other possiblities here, but none of them seemed right.
 
-The third approach, of updating everything in lock-step is what we've
+The third approach, of updating everything in lock-step, is what we've
```
Comment (thejunglejane) [inline docs/maintaining.rst]: Argh sorry. Could this be either "The third approach of updating everything in lock-step is what we've..." or "The third approach, updating everything in lock-step", is what we've"?

of + commas is weird

Comment (leifwalsh) [reply docs/maintaining.rst]: Ok.

**Pull request merged**